app/auth.go

```diff
@@ -208,6 +220,40 @@ func NewAuthManager(conf AuthManagerConfig) *MenderAuthManager {
 	return mgr
 }
 
+// handleCachedTenantToken Handle the cached tenant token if it is not the same
+// as the one in the configuration
+func handleCachedAuthorizationToken(db store.Store, serverURL, tenantToken []byte) error {
```

Comment:
Even though it probably doesn't matter for this specific case, we should always do database updates in transactions, so that there are no race conditions. So we should use `WriteTransaction` at the top here and put the all the code inside, replacing `db` with `txn`.

Reply:
https://github.com/mendersoftware/mender/pull/842/commits/300f183f58745aedd94e2ac2d6fc9cbd9b3934b9
This more like what you had in mind?
I don't really see how else to guard it in a transaction

Reply:
Yes, exactly! This is the same pattern which is used in several other places in the client.